### ISSUE TITLE
Indexing of profiles with the complete time structure

### ISSUE THREAD

**Issue by JulStraus** [sintefore/TimeStruct.jl#31]
At the time being, we only allow the indexing of a `TimeProfile` with a corresponding `TimePeriod` or types like `OperationalScenario`. As a consequence, a user is potentially forced to use a command like

```julia
prof = FixedProfile(10)
ts = SimpleTimes(10, 1)
prof_vect = [prof[t] for t in ts]
```

if they want to receive an array for a given `TimeStructure`. Although it is workable, I would say that `prof[ts]` may be a more elegant solution.

I see one problem related to it as we currently use the `Union` approach:

```julia
function Base.getindex(
    sp::StrategicProfile,
    period::T,
) where {T<:Union{TimePeriod,TimeStructure}}
```

The reasoning for that is that all `AbstractOperationalScenario`, `AbstractRepresentativePeriod`, and so on are `TimeStructure`s. A solution approach would be to provide an intermediate supertype, _e.g._, `TimeStructurePeriod <: TimeStructure`  and then use the function here for this specific type. Note that I have not thought through potential method ambiguities.

**Comment by JulStraus** [comment]
This can be added to #26 or seen as a separate PR.

**Comment by trulsf** [comment]
Fixed by #26
